ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -115,6 +119,8 @@ public void put(MessageWriteSet messageSetToWrite) throws StoreException {
     // Write the blobs in the message set
     CloudWriteChannel cloudWriter = new CloudWriteChannel(this, messageSetToWrite.getMessageSetInfo());
     messageSetToWrite.writeTo(cloudWriter);
+
+    logPutCountsPeriodically();
```

Comment:
I feel like `logPutCountsPeriodically()` depends on `put` method. That is, if `put` method, for some reason, is not invoked then `logPutCountsPeriodically()` won't be called. I am thinking if a background thread to periodically log put counts would be an option. What do you think?

Reply:
That's a good point. While put rate is very small, we may not be able to see the print out, but a background thread is a little bit heavy.

Reply:
Another way (maybe long-term approach) is to implement VCR's own stats manager to collect all statistics we are interested in and report to Helix or back up locally.